src/main/scala/com/gu/ThriftSerializer.scala

```diff
@@ -6,33 +6,46 @@ import com.twitter.scrooge.ThriftStruct
 import org.apache.thrift.protocol.TCompactProtocol
 import org.apache.thrift.transport.TMemoryBuffer
 
-trait ThriftSerializer {
-  import ThriftSerializer._
+object ThriftSerializer {
 
-  def serializeToBytes(struct: ThriftStruct): Array[Byte] = {
-    val buffer = new TMemoryBuffer(ThriftBufferInitialSize)
-    val protocol = new TCompactProtocol(buffer)
-    struct.write(protocol)
-    settings +: payload(buffer.getArray)
-  }
+  private val compressionDefault: CompressionType = NoneType
+  private val initialBufferDefault = 128
+
+  def serializeToBytes(struct: ThriftStruct, userCompressionType: Option[CompressionType], ThriftBufferInitialSize: Option[Int]): Array[Byte] = {
+
+    val buffer = ThriftBufferInitialSize match {
```

Comment:
`ThriftBufferInitialSize` is a value, not a class, it should be lower case.

I think `getOrElse` is easier to read that a `match`, you can write it in one line